tests/unit/Exports/OrderExportTest.php

```diff
@@ -608,6 +609,107 @@ public function testOrderCreateWithEmballage()
         $this->processNewOrder($newOrderId, $newOrder);
     }
 
+    public function testCancelledOrder()
+    {
+        $this->initApiConnection();
+        $this->emptyEnvironment();
+
+        $orderHandler = new OrderHandler();
+
+        $syncedCancelledOrder = WC_Helper_Order::create_order();
+        $syncedCancelledOrder->update_status(OrderExport::STATUS_CANCELLED);
+        $syncedCancelledOrder->save();
+        $orderHandler->addMetadata($syncedCancelledOrder->get_id(), $syncedCancelledOrder);
```

Comment:
what does this do ? Not really clear for me

Reply:
I changed the method name `addMetadata` > `addToBeSynchronizedMetadata`.
But I removed it in this test as upon checking, it was being called during `woocommerce_new_order` hook anyway.